### DISCUSSION TITLE
[symfony/form] Type Error when enity edit with  EnumType Field

### DISCUSSION THREAD

**Discussion by asifkhanpathan** [symfony/symfony#46305] [Q&A]
### Symfony version(s) affected

6.0.8

### Description

when form component  field added as  EnumType  and save to database when we edit same entity then we get type error 
`Symfony\Component\Form\Extension\Core\Type\EnumType::Symfony\Component\Form\Extension\Core\Type\{closure}(): Argument #1 ($choice) must be of type ?BackedEnum, string given, called in /var/www/html/vendor/symfony/form/ChoiceList/ArrayChoiceList.php on line 153`
Field data is null .

### How to reproduce

make Enum Entity 
```
enum Type: string
{
    case red_type = 'red';

    case blue_type = 'blue;



    public function getTypeLabel(): string
    {
        return match ($this) {
            self::red_type => 'Red Color',
            self::blue_type => 'Blue Color',
        };
    }
}
```
then create form Type and enum field
```
$buider ->add('type', EnumType::class, [
                'placeholder' => 'Choose a Type',
                'class' => Type::class,
                'choice_label' => function (?type $type) {
                    return  $type?->getTypeLabel();
                },
                'required' => false,
            ])
```
as we are saving type data as string in database we add transformer
```
      $builder->get('type')->addModelTransformer(new CallbackTransformer(
            function ($type) {
            
                return $type;
            },

            function ($type) {
 
                if (!$type || is_string($type)) {
                    return;
                }
                return $type->getTypeLabel();
            }
        ));
```
this saves entity correctly to database but when same entity we edit we get this error

```
Symfony\Component\Form\Extension\Core\Type\EnumType::Symfony\Component\Form\Extension\Core\Type\{closure}(): Argument #1 ($choice) must be of type ?BackedEnum, string given, called in /var/www/html/vendor/symfony/form/ChoiceList/ArrayChoiceList.php on line 153`
```

**Comment by stof** [comment]
The `choice_label` callback receives the model data. Due to your model transformer, the model data is your string, not your enum.

If your entity contains a string (rather than an enum case managed through the `enumType` feature of doctrine/orm), you should be using the ChoiceType directly instead of using the specialized EnumType subtype.

**Reply by iamkhusainov7** [reply]
@stof Hey :) What if my entity does have Enum type property? I am getting the same issue and I am not able render the form with EnumType :/
